omegaconf/built_in_resolvers.py

```diff
@@ -28,12 +33,22 @@ def decode(expr: Optional[str], _parent_: Container) -> Any:
     return _get_value(val)
 
 
-def dict_keys(in_dict: Dict[Any, Any]) -> List[Any]:
-    return list(in_dict.keys())
+def dict_keys(
+    in_dict: Union[str, Mapping[Any, Any]],
+    _root_: BaseContainer,
+    _parent_: Container,
+) -> ListConfig:
+    return _dict_impl(
+        keys_or_values="keys", in_dict=in_dict, _root_=_root_, _parent_=_parent_
+    )
 
 
-def dict_values(in_dict: Dict[Any, Any]) -> List[Any]:
-    return list(in_dict.values())
+def dict_values(
+    in_dict: Union[str, Mapping[Any, Any]], _root_: BaseContainer, _parent_: Container
+) -> ListConfig:
+    return _dict_impl(
+        keys_or_values="values", in_dict=in_dict, _root_=_root_, _parent_=_parent_
+    )
```

Comment:
What is the desired behavior if there is a missing value in the doct selected from?
I think right now this will raise an exception on the call to access the values.

Reply:
Yeah, I think it's fine, I added tests in 57c3613

We may consider raising an `InterpolationToMissingValueError` instead of a generic `InterpolationResolutionError` wrapping a `MissingMandatoryValue` exception. I can see arguments for either option, so let me know how you feel about this.